### PR TITLE
Fixing some ASATs warning + removed unused class.

### PR DESCRIPTION
**Pull request by ClintonCao**
I've fixed some warnings and I have removed the unused class `CategorySummarizer`.

### CODE REVIEW

**Comment by boninggong** [comment]
Good to see that so much unused code is removed 👍

File: src/main/java/BlueTurtle/summarizers/CategorySummarizer.java [hunk 1]
```diff
@@ -1,98 +0,0 @@
...
- */
-public class CategorySummarizer extends Summarizer {
-
-	@Getter private String category;
-	@Getter private List<Warning> warningList;
-	@Getter private HashMap<String, String> categoryInfo;
-
-	/**
-	 * Constructor.
-	 *
```
Comment (boninggong) [inline src/main/java/BlueTurtle/summarizers/CategorySummarizer.java]: I like how you removed the '*' here

File: src/main/java/BlueTurtle/parsers/GDCParser.java [hunk 2]
```diff
@@ -30,8 +30,7 @@
 	/**
 	 * Constructor. Only through this can GDCParser instantiate itself.
 	 */
-	private GDCParser() {
-	}
+	private GDCParser() { }
```
Comment (boninggong) [inline src/main/java/BlueTurtle/parsers/GDCParser.java]: Good use of spacing and formatting 👍

File: src/main/java/BlueTurtle/gui/GUIController.java [hunk 3]
```diff
@@ -38,10 +38,10 @@
 	public enum ASAT { CheckStyle, PMD, FindBugs; }
 
 	@FXML // ResourceBundle that was given to the FXMLLoader
-	private ResourceBundle resources;
+	private ResourceBundle resources; //NOPMD - needed for the FXMLLoader.
```
Comment (boninggong) [inline src/main/java/BlueTurtle/gui/GUIController.java]: Good to see that comments are added for clarification

File: src/main/java/BlueTurtle/gui/GUIController.java [hunk 4]
```diff
@@ -38,10 +38,10 @@
...
 
 	@FXML // URL location of the FXML file that was given to the FXMLLoader
-	private URL location;
+	private URL location; //NOPMD - needed for the FXMLLoader.
```
Comment (boninggong) [inline src/main/java/BlueTurtle/gui/GUIController.java]: Same as line 41

File: src/main/java/BlueTurtle/gui/GUIController.java [hunk 5]
```diff
@@ -52,7 +52,7 @@
 	@FXML // fx:id="visualizeButton"
 	private Button visualizeButton; // Value injected by FXMLLoader
 	
-	@Getter @Setter private static String projectPath;
+	@Getter @Setter private static String projectPath; //NOPMD - warning caused by lombok.
```
Comment (boninggong) [inline src/main/java/BlueTurtle/gui/GUIController.java]: Ohhh Lombok...

**Comment by TimBuckers** [comment]
Finally, all unnecessary code is removed!

**Pull request merged**